File: AbsManagement_BackEnd/AbsManagementAPI/Core/Models/BaoCaoViPham/BaoCaoViPhamModel.cs
using AbsManagementAPI.Core.Entities;
using AbsManagementAPI.Core.Models.CanBo;

namespace AbsManagementAPI.Core.Models.BaoCaoViPham
{
    public class BaoCaoViPhamModel
    {
        public int Id { get; set; }
        public int? IdCanBoXuLy { get; set; }
        public int? IdBangQuangCao { get; set; }
        public int? IdDiemDatQuangCao { get; set; }
        public string TenHinhThucBaoCao { get; set; }
        public string HoTen { get; set; }
        public string Email { get; set; }
        public string SoDienThoai { get; set; }
        public int IdHinhThucBaoCao { get; set; }
        public string NoiDung { get; set; }
        public string? NoiDungXuLy { get; set; }
        public List<decimal> DanhSachViTri { get; set; }
        public List<string> DanhSachHinhAnh { get; set; }
        public string IdTinhTrang { get; set; }
        public string DiaChi { get; set; }
        public string Phuong { get; set; }
        public string Quan { get; set; }
        public DateTime CreateDate { get; set; }

        public DiemDatQuangCaoEntity DiemDatQuangCao { get; set; }
        public BangQuangCaoEntity BangQuangCao { get; set; }
        //Thông tin cấn bộ
        public string HoTenCanBoXuLy { get; set; }
        public string EmailCanBoXuLy { get; set; }
        public string SoDienThoaiCanBoXuLy { get; set; }
    }
}

File: AbsManagement_FrontEnd/fe-admin/src/apis/baoCaoViPham/baoCaoViPhamModel.ts
import { BangQuangCaoModel } from "../bangQuangCao/bangQuangCaoModel";
import { CanBoModel } from "../canBo/canBoModel";
import { DiemDatQuangCaoModel } from "../diemDatQuangCao/diemDatQuangCaoModel";

interface BaseBaoCaoViPham {
    idCanBoXuLy: number | null;
    idBangQuangCao: number | null;
    idDiemDatQuangCao: number | null;
    tenHinhThucBaoCao: string;
    hoTen: string;
    email: string;
    soDienThoai: string;
    idHinhThucBaoCao : number;
    noiDung: string;
    noiDungXuLy: string | null;
    danhSachHinhAnh: string[];
    danhSachViTri: number[];
    diaChi: string;
    phuong: string;
    quan: string;
    createDate: Date;
    idTinhTrang:string;
    hoTenCanBoXuLy: string;
    emailCanBoXuLy: string;
    soDienThoaiCanBoXuLy:string;
}

export interface BaoCaoViPhamModel extends BaseBaoCaoViPham{
    id: number;
    diemDatQuangCao: DiemDatQuangCaoModel;
    bangQuangCao: BangQuangCaoModel;
    canBoXuLy: CanBoModel | null;
    key:number;
}

export interface CapNhatBaoCaoViPhamModel {
}

export interface ThemMoiBaoCaoViPhamModel extends BaseBaoCaoViPham{
}

export interface XoaBaoCaoViPhamModel{
    id: number;
    ngayCapNhat: Date | null;
}
